samtranslator/translator/translator.py

```diff
@@ -398,6 +399,81 @@ def _get_generated_connector(
         return SamConnector.from_dict(full_connector_logical_id, connector)
 
 
+def traverse_template_depends_on(input_value: Dict[str, Any], resolution_data: Dict[str, str]) -> Dict[str, Any]:
```

Comment:
we already have a traversal  of resources graph, can we use an existing one and just add their what we need to do instead of traversing it again?

Reply:
+1 If we can get all things done in one traversal, that would reduce the overall latency.

Reply:
We can't because the intrinsic resolve is it's own class.